### ISSUE TITLE
Upgrade proto syntax to proto3 [phab:D78]

### ISSUE THREAD

**Issue by kythe** [google/kythe#282]
_Created by **schroederc** at 2015-02-20 20:11:24_: 

Upgrade all proto definitions to use proto3 and update all code to use the new proto3 semantics.

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-02-20 20:41:01_:

Guard against nil nested proto issues in Go

**Comment by kythe** [comment]
_**zarko** wrote at 2015-02-20 20:50:41_:

avast

**Comment by kythe** [comment]
_**zarko** wrote at 2015-02-20 21:41:25_:

  - Update C++ and remove default values from test protos.

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-02-20 21:49:55_:

Split revision into separate commits

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2015-02-20 22:11:51_:

```
if s := req.GetLocation().GetStart(); s != nil {
  windowStart = s.ByteOffset
}
```
et seq.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2015-02-20 22:11:51_:

Needn't be done now, but `strings.Index(v.Signature, vNameFieldSep) != -1` ≣ `strings.Contains(v.Signature, vNameFieldSep)`.

Only saves 3 characters, though. :)

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2015-02-20 22:11:51_:

`name.getSignature().isEmpty()` ?

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2015-02-20 22:11:51_:

+ TODO: Get rid of the `optional` keywords.

**Comment by kythe** [comment]
_**fromberger** wrote at 2015-02-20 22:11:51_:

Just little stuff on the Go side. I didn't look at the C++.

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-02-20 22:28:29_:

Review changes

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-02-20 22:28:36_:

Done

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-02-20 22:28:36_:

Good catch.  Java/Go context switch bug on my part.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-02-20 22:28:36_:

Done

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-02-20 22:28:36_:

Done

**Comment by kythe** [comment]
_**fromberger** wrote at 2015-02-20 23:07:36_:

Looks good from my side. I'll assume Luke will speak up if there's anything odd for C++.

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2015-02-20 23:07:36_

**Comment by kythe** [comment]
_**zarko** wrote at 2015-02-20 23:10:12_:

I contributed the C++ bits so I'm unable to judge them.

**Comment by kythe** [comment]
_**fromberger** wrote at 2015-02-20 23:36:36_:

Good for C++ too. :)
</lazy>

**Comment by kythe** [comment]
_**schroederc** closed via commit df876670029ee6c743631f4099c332f22eb1f25e at 2015-02-20 23:42:02_

**Comment by kythe** [comment]
commit 456255b1c3f5ed290fa547b6d7445325595cda00
78137ada6a2a5f8a33a64e464bc79b4ab809bc9f
07774911ca2cc5033ba624263498c9723e0f22f6
ada304c8856a6ddc50f3ebee86d848b429a80217
